pkg/ebpf/netlink/decoding.go

```diff
@@ -0,0 +1,148 @@
+package netlink
+
+import (
+	"encoding/binary"
+	"net"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	ct "github.com/florianl/go-conntrack"
+)
+
+const (
+	ctaUnspec = iota
+	ctaTupleOrig
+	ctaTupleReply
+)
+
+const (
+	ctaTupleIP    = 1
+	ctaTupleProto = 2
+	ctaTupleZone  = 3
+)
+
+const (
+	ctaIPv4Src = 1
+	ctaIPv4Dst = 2
+	ctaIPv6Src = 3
+	ctaIPv6Dst = 4
+)
+
+const (
+	ctaProtoNum     = 1
+	ctaProtoSrcPort = 2
+	ctaProtoDstPort = 3
+)
+
+var scanner = NewAttributeScanner()
+
+// TODO: Replace the intermediate ct.Con object by the same format we use in the cache
+func DecodeEvent(e Event) []ct.Con {
+	msgs := e.Messages()
+	conns := make([]ct.Con, 0, len(msgs))
+
+	for _, msg := range msgs {
+		c := &ct.Con{}
+		scanner.ResetTo(msg.Data)
+		err := unmarshalCon(scanner, c)
+		if err != nil {
+			log.Debugf("error decoding netlink message: %s", err)
+			continue
+		}
+		conns = append(conns, *c)
+	}
+
+	// Return buffers to the pool
+	e.Done()
```

Comment:
Perhaps we should name this function differently to indicate that it's returning data? I usually assume functions named DecodeX are safe to call more than once